src/messaging/tests/MessagingContext.cpp

```diff
@@ -67,6 +68,16 @@ SessionHandle MessagingContext::GetSessionPeerToLocal()
     return SessionHandle(GetSourceNodeId(), GetPeerKeyId(), GetLocalKeyId(), mDestFabricIndex);
 }
 
+Messaging::ExchangeContext * MessagingContext::NewUnauthenticatedExchangeToPeer(Messaging::ExchangeDelegate * delegate)
+{
+    return mExchangeManager.NewContext(mSecureSessionMgr.CreateUnauthenticatedSession(mPeerAddress).Value(), delegate);
+}
+
+Messaging::ExchangeContext * MessagingContext::NewUnauthenticatedExchangeToLocal(Messaging::ExchangeDelegate * delegate)
```

Comment:
I don't think this will do what we want.

For IP for instance, the full transport context is <protocol, source IP, source port, destination IP, destination port>.

If we want to presume we will always use the same outbound port, we can simplify to <protocol, destination IP, destination port>.  To this end, `chip::Transport::PeerAddress` will work just fine.  However, whether we are the initiator or responder, the session context still should be keyed on peer address, not our local address.

I think `NewUnauthenticatedExchangeToLocal` should be removed and there should only be a single creator method, perhaps called `NewUnauthenticatedExchangeWithPeer`.

Reply:
Loacl and peer are representing two arbitrary nodes in our test context, there is no difference between these two nodes. We may want to rename local and peer to Alice and Bob, to avoid confusion like this.

Reply:
Oh, I see.  Thank you.